src/redux/missions/missions.js

```diff
@@ -1,14 +1,57 @@
+/* eslint-disable no-case-declarations */
 import axios from 'axios';
 
 const GETMISSIONS = 'app/missions/MISSIONS_RETRIEVED';
+const JOINMISSION = 'app/missions/MISSION_JOINED';
+const LEAVEMISSION = 'app/missions/MISSION_LEFT';
 
-const reducer = (missions = [], action) => {
+const reducer = (state = [], action) => {
   const { type, payload } = action;
   switch (type) {
     case GETMISSIONS:
       return [...payload];
+
+    case JOINMISSION:
+      const newMissions = state.map((mission) => {
+        if (mission.id !== payload) {
+          return mission;
+        }
+        return { ...mission, reserved: true };
+      });
+      return [...newMissions];
+    case LEAVEMISSION:
+      const updatedMissions = state.map((mission) => {
+        if (mission.id !== payload) {
+          return mission;
+        }
+        return { ...mission, reserved: false };
+      });
+      return [...updatedMissions];
+
     default:
-      return missions;
+      return state;
+  }
+};
+export const leaveMission = (id) => (dispatch) => {
+  try {
+    dispatch({
+      type: LEAVEMISSION,
+      payload: id,
```

Comment:
Kindly refactor to using action creators according to the ducks pattern.

Reply:
Done, please check